extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReader.java

```diff
@@ -35,16 +37,34 @@ public interface ColumnChunkReader {
     @Nullable
     OffsetIndex getOffsetIndex();
 
+    /**
+     * Used to iterate over column page readers for each page with the capability to set channel context to for reading
+     * the pages.
+     */
+    interface ColumnPageReaderIterator extends Iterator<ColumnPageReader> {
+
+        /**
+         * Set the {@code channelContext} to be used only for a single {@code next()} call. If not set,
+         * {@code SeekableChannelsProvider.ChannelContext.NULL} will be used
+         */
+        void setChannelContext(final SeekableChannelsProvider.ChannelContext channelContext);
```

Comment:
Maybe I was wrong about needing this. I think we want to pass these down on-access, rather than have one permanently linked to a page reader instance.